### PR TITLE
new binding

### PR DESCRIPTION
**Pull request by oblomov-dev**

### CODE REVIEW

**Comment by larshp** [comment]
https://github.com/abaplint/abaplint/pull/3081

File: src/00/z2ui5_cl_fw_utility.clas.abap [hunk 1]
```diff
@@ -508,94 +470,143 @@ CLASS z2ui5_cl_fw_utility IMPLEMENTATION.
...
-    lv_search = get_trim_lower( lv_search ).
+    IF lv_attri_end = `>*`.
+      lv_attri_end = `>`.
+      lv_length = lv_length.
+      lv_attri = lv_attri(lv_length) && lv_attri_end.
+    ELSE.
+      lv_attri_end = `-`.
+      lv_length = lv_length + 2.
+      lv_attri = lv_attri(lv_length) && lv_attri_end.
+    ENDIF.
```
Comment (larshp) [inline src/00/z2ui5_cl_fw_utility.clas.abap:505]:
```suggestion
    IF lv_attri_end = `>*`.
      lv_attri_end = `>`.
      lv_length = lv_length.
    ELSE.
      lv_attri_end = `-`.
      lv_length = lv_length + 2.
    ENDIF.
    lv_attri = lv_attri(lv_length) && lv_attri_end.    
```

Comment (oblomov-dev) [reply src/00/z2ui5_cl_fw_utility.clas.abap]: thanks, just sow it now and added it.
working on a bigger change here, still need to review and fix the code a few more times.

**Comment by larshp** [comment]
update to latest in package.json, https://github.com/abaplint/abaplint/pull/3083 contains a downport fix

**Comment by oblomov-dev** [comment]
@larshp updated, thank you, downport works now!

**Comment by larshp** [comment]
a fix to `replace()` is part of https://github.com/abaplint/transpiler/pull/1302

**Comment by oblomov-dev** [comment]
ah great!

**Comment by oblomov-dev** [comment]
is there also the chance to get a uuid length 22?
https://github.com/abap2UI5/abap2UI5/actions/runs/5962212278/job/16172965213?pr=503

can not use uuid32, can't send it from front to backend because /ui2/json only allows attributes length up to 30 characters.

**Comment by larshp** [comment]
to be added to https://github.com/open-abap/open-abap-core/blob/main/src/uuid/cl_system_uuid.clas.abap

**Comment by larshp** [comment]
try again, https://github.com/open-abap/open-abap-core/pull/760 merged

**Comment by oblomov-dev** [comment]
ah interesting to see how this works! maybe in the future i can do my own fixes. using abap to create javascript is always exciting :)

the bug is fixed! there are and i expect also a few more. i had to make a bigger change to make the frontend model creation more generic and cleaned up. Maybe i refactor and finish my changes first before trying to find out where exactly open-abap is struggeling and then i can try to adjust my code.

**Comment by larshp** [comment]
https://github.com/open-abap/open-abap-core/pull/762

**Comment by oblomov-dev** [comment]
as always, thank you! json works now.
the good news, i added a lot more unit tests, the bad news i expect some more bugs. if there is something i can do on my side, let me know.

**Pull request merged**